### DISCUSSION TITLE
How to retrieve DiscordConnection from a Member or User

### DISCUSSION THREAD

**Discussion by andrewfluck** [kordlib/kord#851] [Help] [answered]
I looked around a bit and couldn't find any references to DiscordConnection other than the class itself. Is this not implemented yet?

**Comment by DRSchlaubi** [comment, marked as answer]
DiscordConnection is part of the OAuth API so it can't be obtained through Kord

**Reply by andrewfluck** [reply]
What about for a Linked Role with a connection to Twitter? The use case is automating a roster list/spreadsheet for known Twitter subs who join.

**Reply by DRSchlaubi** [reply]
You can only publish the bot related api, which is publishing the required metadata

**Reply by andrewfluck** [reply]
Gotcha, so this might not be possible to do automatically without having a command for a user to run, then

**Reply by lukellmann** [reply]
you need some sort of explicit opt-in by the user.

one way to do it is to complete an oauth flow to get an access token for each user to then [get the user connections](https://discord.com/developers/docs/resources/user#get-user-connections). as @DRSchlaubi said, this isn't supported by kord, so you would have to implement it on your own.

the probably easier way to do it is to set up a linked role in the guilds you want to have this for. users will then opt in to that role and you can use a bot in the guild to see which users have the role. however you won't be able to [get the information associated with the connected application](https://discord.com/developers/docs/resources/user#get-user-application-role-connection) (like the twitter username) as that would require an oauth token for the application (so you would have to be twitter in this case).
